cli/cmd/lql.go

```diff
@@ -19,71 +19,217 @@
+		lqlValidateCmd,
 	}
 )
 
 func init() {
 	// add the lql command
 	rootCmd.AddCommand(lqlCmd)
 
-	// file flag to specify a query from disk
-	lqlCmd.Flags().StringVarP(&lqlFile,
-		"file", "f", "",
-		"path to an LQL query to run",
+	// add sub-commands to the lql command
+	lqlCmd.AddCommand(lqlRunCmd)
+
+	// for commands that take query "text" as input
+	for _, cmd := range lqlQueryCommands {
+		// file flag to specify a query from disk
+		cmd.Flags().StringVarP(
+			&lqlFile,
+			"file", "f", "",
+			"path to an LQL query to run",
+		)
+		// repo flag to specify a query from repo
+		cmd.Flags().BoolVarP(
+			&lqlRepo,
+			"repo", "r", false,
+			"id of an LQL query to run via active repo",
+		)
+		// url flag to specify a query from url
+		cmd.Flags().StringVarP(
+			&lqlURL,
+			"url", "u", "",
+			"url to an LQL query to run",
+		)
+	}
+
+	// run specific flags
+	// env flag to specify a query from disk
+	lqlRunCmd.Flags().BoolVarP(
+		&lqlEnv,
+		"env", "e", false,
+		"run an LQL query by ID (using active profile)",
+	)
+	// start time flag
+	// TODO: come up with reasonable default per UI (1d)
+	lqlRunCmd.Flags().StringVarP(
+		&lqlStart,
+		"start", "", "",
+		"start time for LQL query",
+	)
+	// end time flag
+	// TODO: come up with reasonable default per UI (1d)
+	lqlRunCmd.Flags().StringVarP(
+		&lqlEnd,
+		"end", "", "",
+		"end time for LQL query",
+	)
+	lqlRunCmd.Flags().BoolVarP(
+		&lqlValidate,
+		"validate_only", "", false,
+		"validate query only (do not run)",
 	)
 }
 
-func runLQLQuery(_ *cobra.Command, args []string) error {
-	var query = ""
+// for commands that take a query as input
+func inputQuery(cmd *cobra.Command, args []string) (
+	query string,
+	err error,
+) {
+	var queryID string
 
+	// if an inline argument was provided
+	// determine if it's a query or a query identifier
 	if len(args) != 0 && args[0] != "" {
-		query = args[0]
+		if lqlEnv || lqlRepo {
+			queryID = args[0]
+		} else {
+			query = args[0]
+		}
+	}
+
+	if lqlEnv {
+		var queryResponse api.LQLQueryResponse
+		queryResponse, err = cli.LwApi.LQL.GetQueryByID(queryID)
+		if err == nil && len(queryResponse.Data) != 0 {
+			query = queryResponse.Data[0].QueryText
+		}
+	} else if lqlRepo {
+		err = errors.New("NotImplementedError")
 	} else if lqlFile != "" {
-		lqlQuery, err := ioutil.ReadFile(lqlFile)
+		var fileData []byte
+		fileData, err = ioutil.ReadFile(lqlFile)
 		if err != nil {
-			return errors.Wrap(err, "unable to read file")
+			err = errors.Wrap(err, "unable to read file")
+			return
 		}
-		query = string(lqlQuery)
+		query = string(fileData)
+	} else if lqlURL != "" {
+		msg := "unable to open URL"
+		var response *http.Response
+		var body []byte
+
+		response, err = http.Get(lqlURL)
+		if err != nil {
+			err = errors.Wrap(err, msg)
+			return
+		}
+		defer response.Body.Close()
+
+		if response.StatusCode != 200 {
+			err = errors.Wrap(errors.New(response.Status), msg)
+			return
+		}
+
+		body, err = io.ReadAll(response.Body)
```

Comment:
ReadAll is from the ioutil package

```suggestion
		body, err = ioutil.ReadAll(response.Body)
```

Reply:
Good eye.  No idea why this actually works...

Reply:
Fixed.